### DISCUSSION TITLE
Can see private google calendar events on default calendar module but not calendarext3

### DISCUSSION THREAD

**Discussion by Ashna-git-hub** [MMRIZE/MMM-CalendarExt3#162] [Q&A]
Can someone upload an example of their working calendar config when using a Google calendar.  It works on normal calendar but not this module calendarext3. I can see the public holidays.  I can see my events on default module.

I am using the secret url and it works

**Comment by eouia** [comment]
Try this;
```js
{
	module: "calendar",
	hiddenOnStartup: true,
	position: "top_left",
	config: {
		calendars: [
			{
				symbol: "calendar-check",
				url: "https://ics.calendarlabs.com/76/mm3137/US_Holidays.ics"
			},
			{
				color: "gold",
				symbol: "wrench",
				url: "https://calendar.google.com/calendar/ical/4b9cf164fc89b0242c19954ce50bee32d6b11f22867bca27dd4bae74c9ce6390%40group.calendar.google.com/private-e481c2eeb134dd8eec7a4eec9322c7cc/basic.ics"
			},
		]
	}
},
{
	module: "MMM-CalendarExt3",
	position: "bottom_bar",
	config: {
		mode: "monthly",
	}
},
```

It will show this;
![image](https://github.com/user-attachments/assets/5533a79e-c6f3-4b6c-bee1-fea9b59213da)

**Reply by Ashna-git-hub** [reply]
Thank you for this I gave it a go and unfortunately it's still not working, when I reveal the normal calendar my events show

**Comment by Ashna-git-hub** [comment]
![20240827_123903](https://github.com/user-attachments/assets/38f270e6-dcce-4bc4-97e2-bb5bf421b16f)
It is still not working

**Comment by eouia** [comment]
Can you send me your config.js to me? (eouia0819@gmail.com)

**Reply by Ashna-git-hub** [reply]
All sent thank you so much

**Reply by eouia** [reply]
![image](https://github.com/user-attachments/assets/e69e05ee-3f6a-451c-9ab9-27b81ef8ae7f)
It shows like this. nothing wrong. (Only MMM-Globe, I have not installed, so that is not shown.)

**Reply by Ashna-git-hub** [reply]
Hmmm do you think it's not showing on my screen due to screen rotation? How long did the events take to load?
It's still not working on mine

**Reply by Ashna-git-hub** [reply]
Hello this works for me now I let it load but the calendar fetcher runs 5 mins after I press run. Therefore, I only see events after 5 mins. Is there a way to force this ?

**Reply by eouia** [reply]
Now I know what happened to you.

Even if some calendars haven't finished fetching by then, CX3 starts rendering the calendars on the screen after a certain amount of time has passed since the last calendar was fetched, even if not all calendars have been fetched yet, in order to prevent waiting for too long or infinitely.

Calendars that haven't been fetched by then will probably be fetched in the meantime, so they will be displayed on the screen at the next cycle's rendering time. Of course, there may be calendars that are never fetched for some reason, but this won't interrupt the rendering.

However, in this case, some calendars may not be visible yet during the first rendering cycle. This usually happens when not all calendars have been fetched within `waitFetch`, for example, on a device with low computing power, slow internet speed, too many events in the calendar, or the calendar fetching schedule is too long.

There are several options:

1) Give `waitFetch` a sufficiently long time. This property specifies a certain amount of time to wait after the last calendar has been fetched before it is reflected on the screen. This delays the rendering until other calendars are fetched.

2) Shorten `refreshInterval`. The calendars will probably be fetched by the next rendering cycle, so if you reflect them on the screen more frequently, they will be displayed on the screen faster.

3) However, my recommendation is to just wait. Since some slow calendars are not visible in the first rendering cycle anyway, they will be reflected after 5 minutes (default refreshInterval). This problem will be solved automatically after a while without having to adjust waitFetch or refreshInterval.

**Reply by Ashna-git-hub** [reply]
Thank you so much for this. We have a ton of events in our calendar and I can see this fill up the log in the DEBUG events so I imagine it is this. Really appreciate your response and help this is all really good.

**Reply by Kaushagen** [reply]
Sorry to bring the subject alive again, but I have a similar problem - kind of.
Dafult calendar setting ok, pointing to norwegian google holiday calendar. But enterering new calendar module ((CalExt3) and settings for my personal google calendar it just shows a basuc google calendar and only three seeks …!?

**Reply by Kaushagen** [reply]
I forgot to include my config file
[MMM-CalExt3 Mine settinger.docx](https://github.com/user-attachments/files/21823547/MMM-CalExt3.Mine.settinger.docx)
